File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/KnowledgeCategory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities
{
    public class KnowledgeCategory : Entity
    {
        public Guid KnowledgeCategoryId { get; set; }
        public Guid KnowledgeId { get; set; }
        public Guid CategoryId { get; set; }
        protected override KnowledgeCategory Init()
        {
            return new KnowledgeCategory();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/entities/knowledge-category.js

import Entity from './entity';

class KnowledgeCategory extends Entity {
    constructor() {
        super();
        this.knowledgeCategoryId = null;
        this.knowledgeId = null;
        this.categoryId = null;
    }

    init() {
        return new KnowledgeCategory();
    }
}

export default KnowledgeCategory;